test/test_images/probe_helper/probe/broker_e2e_delivery.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"context"
+	"fmt"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	cecontext "github.com/cloudevents/sdk-go/v2/context"
+)
+
+const (
+	// BrokerE2EDeliveryProbeEventType is the CloudEvent type of broker e2e
+	// delivery probes.
+	BrokerE2EDeliveryProbeEventType = "broker-e2e-delivery-probe"
+
+	brokerExtension    = "broker"
+	namespaceExtension = "namespace"
+)
+
+// BrokerE2EDeliveryProbe is the probe handler for probe requests in the broker
+// e2e delivery probe.
+type BrokerE2EDeliveryProbe struct{}
+
+var brokerE2EDeliveryProbe Handler = &BrokerE2EDeliveryProbe{}
+
+func (p *BrokerE2EDeliveryProbe) Forward(ctx context.Context, ph *Helper, event cloudevents.Event) error {
+	namespace, ok := event.Extensions()[namespaceExtension]
+	if !ok {
+		return fmt.Errorf("Broker e2e delivery probe event has no '%s' extension", namespaceExtension)
+	}
+	broker, ok := event.Extensions()[brokerExtension]
+	if !ok {
+		broker = "default"
+	}
+
+	// Get the receiver channel
+	channelID := fmt.Sprintf("%s/%s", event.Extensions()[probeEventTargetServiceExtension], event.ID())
```

Comment:
nit: Let's make channelID a helper function to guarantee the same channelID implementation in both Forward and Receive.

Reply:
Done.